apps/shell/app.js

```diff
@@ -23,6 +25,161 @@ if (fs.existsSync('.env')) {
   dotenv.config({path: '.env'});
 }
 
+//Start Terminal Color Scheme Implementation
+//declared constants for directory paths
+const xdg_config_dir = (process.env["XDG_CONFIG_HOME"] || path.join(os.homedir(), ".config"));
+const ood_app_config_root = (process.env["OOD_APP_CONFIG_ROOT"] || '/etc/ood/config/ondemand/apps/shell');
+const userDir = path.join(xdg_config_dir, "ondemand", "apps", "shell", "themes");
+const systemDir = path.join(ood_app_config_root, "themes");
+
+//declared global variables for color schemes
+var schemeObjects;
+
+checkDirSync(userDir);
+checkDirSync(systemDir);
+
+if (process.env["SYSTEM_SCHEMES"] === true) {
+    var user = getSchemeObjects(userDir);
+    var system = getSchemeObjects(systemDir);
+
+   schemeObjects = {...getSchemeObjects(userDir), ...getSchemeObjects(systemDir)};
+
+} else {
+   schemeObjects = {...getSchemeObjects(userDir)};
+}
+
+// helper functions
+function checkDirSync(dir) {  
+  try {
+    fs.statSync(dir);
+  } catch(e) {
+    if (dir === systemDir) {
+      process.env["SYSTEM_SCHEMES"] = false;
+    } else if (dir === userDir) {
+      fs.mkdirSync(dir);
+    }
+  }
+}
+
+function getSchemeObjects(dir) {
```

Comment:
If themes were stored in an array instead of as an object, the code may be easier to reason about. You would then use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find to get the theme by filename.